scijava/scijava-ops-engine/src/main/java/org/scijava/ops/engine/DefaultOpEnvironment.java

```diff
@@ -166,30 +194,32 @@ public static List<MatchingRoutine> getMatchingRoutines(
 	}
 
 	@Override
-	public Set<OpInfo> infos() {
+	public List<OpInfo> infos() {
 		if (opDirectory == null) initOpDirectory();
-		return opDirectory.values().stream().flatMap(Collection::stream).collect(
-			Collectors.toSet());
+		// NB distinct() prevents aliased Ops from appearing multiple times.
+		return opDirectory.values().stream() //
+				.flatMap(Collection::stream).distinct().sorted()
+				.collect(Collectors.toList());
 	}
 
 	@Override
-	public Set<OpInfo> infos(String name) {
+	public List<OpInfo> infos(String name) {
 		if (opDirectory == null) initOpDirectory();
 		if (name == null || name.isEmpty()) return infos();
 		return opsOfName(name);
 	}
 
 	@Override
-	public Set<OpInfo> infos(Hints hints) {
+	public List<OpInfo> infos(Hints hints) {
 		return filterInfos(infos(), hints);
 	}
 
 	@Override
-	public Set<OpInfo> infos(String name, Hints hints) {
+	public List<OpInfo> infos(String name, Hints hints) {
 		return filterInfos(infos(name), hints);
 	}
 
-	private Set<OpInfo> filterInfos(Set<OpInfo> infos, Hints hints) {
+	private List<OpInfo> filterInfos(List<OpInfo> infos, Hints hints) {
```

Comment:
Here's a place where the input `infos` could be a `Collection`, because all you do with them is call `.stream()`. So you should type the method argument directly on the interface that declares the `stream()` method.

Reply:
That's true, it could, but it's also a private method, and it makes things simpler within the public functions, so I'd say we leave it as is?